src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/matcher/OffenderMatcher.kt

```diff
@@ -4,40 +4,20 @@ import uk.gov.justice.digital.hmpps.personrecord.client.model.OffenderDetail
 import uk.gov.justice.digital.hmpps.personrecord.model.Person
 import uk.gov.justice.digital.hmpps.personrecord.validate.PNCIdentifier
 
-class OffenderMatcher(val offenderDetails: List<OffenderDetail>?, val person: Person) {
-
-  fun isExactMatch(): Boolean {
-    return !offenderDetails.isNullOrEmpty() && offenderDetails.size == 1 && offenderDetails.any { isMatchingOffender(person, it) }
-  }
-
-  fun isPartialMatch(): Boolean {
-    return !offenderDetails.isNullOrEmpty() && offenderDetails.size == 1 && offenderDetails.any {
-      person.otherIdentifiers?.pncIdentifier == PNCIdentifier(it.otherIds.pncNumber) &&
-        it.firstName.equals(person.givenName, true) || it.surname.equals(person.familyName, true) || it.dateOfBirth == person.dateOfBirth
-    }
-  }
-
-  fun isMultipleMatch(): Boolean {
-    return !offenderDetails.isNullOrEmpty() && offenderDetails.size > 1 && offenderDetails.all { isMatchingOffender(person, it) }
-  }
-
+class OffenderMatcher(offenderDetails: List<OffenderDetail>?, person: Person) :
+  Matcher<OffenderDetail>(offenderDetails, person) {
   fun isPncDoesNotMatch(): Boolean {
-    return !offenderDetails.isNullOrEmpty() && offenderDetails.none {
-      person.otherIdentifiers?.pncIdentifier == PNCIdentifier(it.otherIds.pncNumber)
+    return !items.isNullOrEmpty() && items.none {
+      person.otherIdentifiers?.pncIdentifier == PNCIdentifier(it.otherIds.pncNumber.toString())
```

Comment:
`it.otherIds.pncNumber` is already a `String?` - no need for the toString() call

Reply:
I think actually making IDs.pncNumber into a PNCIdentifier is probably the right way to do things, but that is not necessary for this PR :-)